### ISSUE TITLE
Move silhouette/iframe logic before dojo parser?

### ISSUE THREAD

**Issue by JonFerraiolo** [maqetta/maqetta#925]
@jhpedemonte, @billreed63, @peller

In looking at code flow paths in the debugger, I see that dojox.mobile does some of its layout at parse time, and we don't set up the device silhouette and inner iframe dimensions until after parse time. As a result, in some cases, widgets are mis-positioned because the layout calculations used the iframe dimensions at parse time, which are different than the iframe dimensions when the iframe is (visually) sitting inside the silhouette.

I wouldn't be surprised if we have some layout/resize/render bugs because of this. There is also an issue where the parser is called before the editor toolbar is completely rendered, so the iframe at parse time has a larger height then after the toolbar is completed.

The obvious thing to try is to move the silhouette and iframe sizing logic before calling the Dojo parser.

**Comment by peller** [comment]
It is a bug if the widgets do not respond properly to a page resize.  Perhaps mobile being more 'lightweight' is not registering enough event handlers as dijit might, but consider an orientation change.  Sounds like a good idea to establish our UI prior the parse, regardless, to avoid screen redraws.

**Comment by JonFerraiolo** [comment]
Remember, in the page editor, we disable most of the runtime JavaScript. As a result, some things don't resize automatically like they would if running directly in browser (i.e., not in Maqetta page editor).

But if we don't notice problems before M4 ships, then probably best to defer this to M5. I suggest leaving this open for M4 for awhile as a reminder for more testing.

**Comment by peller** [comment]
aside from the div blocking click events, do we really disable that much widget JS?  I know the Dijit layout widgets retain much of their quirky resize behaviors.  I wouldn't dream of messing with this stuff until M5, though.

**Comment by JonFerraiolo** [comment]
We remove all SCRIPT tags and on**\* attributes from the user's HTML file. The page editor then dynamically adds various SCRIPT tags (particularly for toolkits such as Dojo) and dojo.require() as it parses the page and the user adds widgets to the page. So, it's a complex situation.

**Comment by peller** [comment]
true... for user code. but the sizing behavior and most other behavior is baked into the widget and remains active.

**Comment by JonFerraiolo** [comment]
I'm not aware of any bad bugs due to this issue, so moving to Priority-Low for time being so we don't forget about it but also so that if we don't see any bad bugs during further testing, then move to M5.

**Comment by JonFerraiolo** [comment]
Pushing to M6

**Comment by JonFerraiolo** [comment]
-> M7

**Comment by JonFerraiolo** [comment]
-> M8
